### PR TITLE
remove full iface access assumption [linux/bsd]

### LINKED ISSUE #8: We shouldn't make assumptions about interface indices
```
    TestRoute: netroute_test.go:12: out of order iface 1 = {6436 1500 eth0 02:42:ac:1c:00:03 up|broadcast|multicast}
```

### PR DESCRIPTION
**Pull request by willscott**
fix #8

### CODE REVIEW

**Comment by Stebalien** [comment]
Ok, well, upgrading something along the way has exposed the actual error. We're getting an index out of bounds error.

**Comment by Stebalien** [comment]
```
Failed
=== RUN   TestRoute
--- FAIL: TestRoute (0.00s)
panic: runtime error: index out of range [4619] with length 2 [recovered]
	panic: runtime error: index out of range [4619] with length 2

goroutine 6 [running]:
testing.tRunner.func1.1(0x5447e0, 0xc000016920)
	/usr/local/go/src/testing/testing.go:941 +0x3d0
testing.tRunner.func1(0xc000022240)
	/usr/local/go/src/testing/testing.go:944 +0x3f9
panic(0x5447e0, 0xc000016920)
	/usr/local/go/src/runtime/panic.go:967 +0x166
github.com/libp2p/go-netroute.(*router).RouteWithSrc(0xc0000ac1e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00001a3d0, 0x10, 0x10, ...)
	/home/circleci/project/common.go:103 +0x7e2
github.com/libp2p/go-netroute.(*router).Route(0xc0000ac1e0, 0xc00001a3d0, 0x10, 0x10, 0x10, 0x10, 0x1, 0x0, 0xc0000ea358, 0x4, ...)
	/home/circleci/project/common.go:82 +0x96
github.com/libp2p/go-netroute.TestRoute(0xc000022240)
	/home/circleci/project/netroute_test.go:51 +0x4af
testing.tRunner(0xc000022240, 0x55ff40)
	/usr/local/go/src/testing/testing.go:992 +0xdc
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1043 +0x357
```

**Comment by Stebalien** [comment]
We probably need to use a map instead of an array.

**Comment by willscott** [comment]
tests passing is good. I'll merge & cut a release now to not block your downstream work. Take a look and confirm when you get a chance.

File: netroute_linux.go [hunk 1]
```diff
@@ -23,6 +22,8 @@ import (
 
 func New() (routing.Router, error) {
 	rtr := &router{}
+	rtr.ifaces = make(map[int]net.Interface)
+	rtr.addrs = make(map[int]ipAddrs)
```
Comment (Stebalien) [inline netroute_linux.go]: nit: We should probably pre-allocate these.

**Pull request merged**